### PR TITLE
Subscription management: Apply updated design to mobile header

### LINKED ISSUE #74526: `/settings`: Apply updated design to mobile header
## Design
inDLaEQV8jJ21O4WXawIsJ-fi-712-16233
Image

## Related conversation
p1678388803357779-slack-C02TCEHP3HA

___
Thanks, @saygunnyc, for a quick turnaround on this! 🙇‍♂️

### PR DESCRIPTION
**Pull request by phcp**
Closes [#74526](https://github.com/Automattic/wp-calypso/issues/74526)

## Proposed Changes

* Apply the updated design to mobile header
* Hide the floating menu
* Hide the WordPress blue text from the logo on small screen sizes (< 480px)
* Update nav CTA button padding to match the Figma design
* Update title font size to match the Figma design
* Increase space between content and header to match the Figma design

Screenshot 2023-03-21 at 18 07 09

## Testing Instructions

* Run this patch on your local env
* Go to http://calypso.localhost:3000/subscriptions/settings
* Verify if the header is rendered as expected on small screen sizes
* Verify if no regression was made on bigger sizes

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by ivan-ottinger — APPROVED** [review]
Looks great! 👍🏼👍🏼 Just a couple things I noticed:

When looking at the mobile screen as a logged-in user, the heading is shifted too much to the top (related to @yansern comment here: p1679475669380209/1679474258.545539-slack-C02TCEHP3HA). This hasn't been introduced by this PR and is in the current `trunk` already:

Markup on 2023-03-22 at 10:22:02

---  

> Hide the WordPress blue text from the logo on small screen sizes (< 480px)

I see this is not a part of the Figma design (inDLaEQV8jJ21O4WXawIsJ-fi-1370-17981), but I guess it was necessary to remove the WordPress.com text on those small screens, right?

---

I have approved the PR as I think it is good to go - if we decide to address the above issues in a separate PRs.

**Comment by phcp** [comment]
> When looking at the mobile screen as a logged-in user, the heading is shifted too much to the top (related to @yansern comment here: p1679475669380209/1679474258.545539-slack-C02TCEHP3HA). This hasn't been introduced by this PR and is in the current trunk already

@ivan-ottinger True, this only covers external user scenarios, ~~I'll open a task to add the header for logged in users.~~ It'll also cover the header layout for logged-in users after this PR be merged https://github.com/Automattic/wp-calypso/pull/74733

> I see this is not a part of the Figma design (inDLaEQV8jJ21O4WXawIsJ-fi-1370-17981), but I guess it was necessary to remove the WordPress.com text on those small screens, right?

It's an updated version of the mobile header, you can see it here on Figma inDLaEQV8jJ21O4WXawIsJ-fi-712-13475

**Comment by phcp** [comment]
~~On hold by https://github.com/Automattic/wp-calypso/pull/74733/~~
Unblocked by p1679508249629529/1679500749.842179-slack-C02TCEHP3HA

**Review by JuanLucha — APPROVED** [review]

**Pull request merged**
